go/libraries/doltcore/sqle/dsess/session_state_adapter.go

```diff
@@ -120,6 +124,15 @@ func (s SessionStateAdapter) GetRemotes() (map[string]env.Remote, error) {
 	return s.remotes, nil
 }
 
+func (s SessionStateAdapter) GetBranches() (map[string]env.BranchConfig, error) {
+	return s.branches, nil
+}
+
+func (s SessionStateAdapter) UpdateBranch(name string, new env.BranchConfig) error {
+	s.branches[name] = new
```

Comment:
Probably should throw an error, as this doesn't get persisted anywhere

Reply:
`--set-update` works as expected within in a session, but loses state on restart. do you want to disable it?